tests/test_corr.py

```diff
@@ -0,0 +1,126 @@
+
+    return pd.DataFrame(
+        data=random_data,
+        index=[f"ENSG00000123456.{i}" for i in range(random_data.shape[0])],
+        columns=[f"Sample-Number-{i}" for i in range(random_data.shape[1])],
+    )
+
+
+def _run_basic_checks(corr_method, random_state=0) -> tuple[pd.DataFrame, pd.DataFrame]:
+    """
+    Performs basic checks on the output of a correlation method.
+
+    Args:
+        corr_method: a function that computes the correlations among genes. This
+            function receives the data as the only arguments, which has the same
+            format returned by function _get_random_data. It must return a
+            correlation matrix as with the same format specified in the corr.py
+            module (description at the top of file).
+        random_state: passed to the _get_random_data function.
+
+    Returns:
+        A tuple with the random data generated and the correlation matrix.
+    """
+    n_genes = 10
+    n_samples = 100
+    random_data = _get_random_data(n_genes, n_samples, random_state)
+
+    # run
+    corr_mat = corr_method(random_data)
+    assert corr_mat is not None
+
+    # shape
+    assert hasattr(corr_mat, "shape")
+    assert corr_mat.shape == (n_genes, n_genes)
+
+    # row/columns names are genes
+    assert hasattr(corr_mat, "index")
+    assert list(corr_mat.index) == list(random_data.index)
+
+    assert hasattr(corr_mat, "columns")
+    assert list(corr_mat.columns) == list(random_data.index)
+
+    assert hasattr(corr_mat, "columns")
+
+    # dtype
+    assert np.issubdtype(corr_mat.to_numpy().dtype, np.number)
+
+    # diagonal has ones
+    assert np.array_equal(np.diag(corr_mat), np.ones(n_genes))
+
+    # matrix is symmetric
+    assert corr_mat.equals(corr_mat.T)
+
+    return random_data, corr_mat
+
+
+def test_corr_pearson():
+    # run basic tests first
+    data, corr_mat = _run_basic_checks(corr.pearson)
+
+    corr_values = pd.Series(corr_mat.to_numpy().flatten())
+
+    # check ranges
+    assert corr_values.max() <= 1.0
+    assert corr_values.min() >= -1.0
+    assert np.sign(corr_values.max()) != np.sign(corr_values.min())
+
+    # calculate pearson with a different method and check if it is the same
+    numpy_pearson_mat = np.corrcoef(data.to_numpy())
+
+    assert np.allclose(
+        numpy_pearson_mat,
+        corr_mat.to_numpy(),
+    )
+
+
+def test_corr_spearman():
+    # run basic tests first
+    data, corr_mat = _run_basic_checks(corr.spearman)
+
+    corr_values = pd.Series(corr_mat.to_numpy().flatten())
+
+    # check ranges
+    assert corr_values.max() <= 1.0
+    assert corr_values.min() >= -1.0
+    assert np.sign(corr_values.max()) != np.sign(corr_values.min())
+
+    # calculate pearson with a different method and check if it is the same
+    from scipy.stats import spearmanr
+
+    scipy_spearman_mat = spearmanr(data.to_numpy(), axis=1)[0]
+
+    assert np.allclose(
+        scipy_spearman_mat,
+        corr_mat.to_numpy(),
+    )
```

Comment:
To be super safe, you could add test a 2x2 matrix where you've calculated the correlations by hand just to make sure that the numpy and scipy correlation functions are behaving like you'd expect them to. It's probably fine though

Reply:
Good point, you are right. I think it's good to make the test as simple as possible, and that includes using constants, not computed variables within the test. I've added a couple of new unit tests with this.